tools/linter/clang_tidy/__main__.py

```diff
@@ -68,7 +71,13 @@ def parse_args() -> argparse.Namespace:
         help="Path to the folder containing compile_commands.json",
     )
     parser.add_argument(
-        "--diff-file", help="File containing diff to use for determining files to lint and line filters"
+        "--diff-file",
+        help="File containing diff to use for determining files to lint and line filters",
+    )
+    parser.add_argument(
```

Comment:
I'd prefer if we kept `actions_local_runner.py` as a main dispatcher for local lints so we can avoid duplicating stuff like `--changed-only`, it could dispatch out to this script but could you add it as an explicit job here?

https://github.com/pytorch/pytorch/blob/master/tools/actions_local_runner.py#L407

If you do that and implment `quick()`, `full()`, and `filter_files()` this should all come for free. It should work to pass the files list through manually with the `--path` option here so we don't have to add any new ones.